rest/client/rest.go

```diff
@@ -1215,3 +1215,55 @@ func (c *communicatorImpl) GetHostProcessOutput(ctx context.Context, hostProcess
 
 	return result, nil
 }
+
+func (c *communicatorImpl) GetS3TaskCredentials(ctx context.Context) (*evergreen.S3Credentials, error) {
+	info := requestInfo{
+		method:  http.MethodGet,
+		version: apiVersion2,
+		path:    "/task/s3_read_credentials",
+	}
+
+	resp, err := c.request(ctx, info, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't make request to get task read-only credentials")
+	}
```

Comment:
shuld probably defer close body here?

Reply:
Technically, `util.ReadJSONInto` closes it, but I closed anyways.